app/controllers/profiles_controller.rb

```diff
@@ -19,12 +34,116 @@ def create
       redirect_to root_url
     else 
       render 'new'
+      
     end
   end
 
+  
+
   private
 
   def user_params
     params.require(:user).permit(:name, :email, :password, :password_confirmation)
   end
+
+
+  def verify_summoner_name
+    begin
+      @summonerReq = Lol::SummonerRequest.new Rails.application.secrets.sulai_api_key, "na"
+      @summonerObj = @summonerReq.by_name(user_params[:name]).first
+    rescue Lol::NotFound => e
+      if e.message == '404 Not Found'
+        flash[:danger] = "Summoner name has to be registered. Register on the LoL website and return here to sign up"
+        redirect_to new_profile_path
+      end
+    end
+  end
+
+  def get_summoner
+    @summonerReq = Lol::SummonerRequest.new Rails.application.secrets.sulai_api_key, "na"
+    @summonerObj = @summonerReq.by_name(User.find(params[:id]).name).first
+  end
+  
+  #Static Request is used to retrieve info on any of champion, item, mastery, rune summoner_spell
+  def get_champion_request
+    @champsReq = Lol::StaticRequest.new Rails.application.secrets.sulai_api_key, "na"
+  end
+
+  def get_summoner_ranked_stats
+    @statsReq = Lol::StatsRequest.new Rails.application.secrets.sulai_api_key, "na"
+    @statsObj = @statsReq.ranked(@summonerObj.id, extra = {})
+    #Since the champion with id = 0 always returns 404 error and is used for error checking, I am deleting it from the array.
+    @statsObj.champions.delete_if{ |h| h.id == 0 }
+  end
+
+  def get_most_played_champ
+    #Most played champion is returned by sorting by total sessions played descending and returning second champion.
+    #First champion does not exist - refer below
+    #Most summoners I have tried have ID = 0 for champions with the highest total session played.
+    #However, I had trouble retrieving this champion and after researching for a while, I found that champion with ID = 0 does not exist and
+    #is only used to deal with exceptions. Linked here - https://market.mashape.com/community/elophant-league-of-legends
+    @mostPlayedChampObj = @statsObj.champions.sort_by{|x| -x.stats.total_sessions_played}
+    @mostPlayedChampObj = @mostPlayedChampObj.paginate(:page => params[:page], :per_page => 5)
+  end
+
+  def get_overall_WLR
+    #OverallWLR
+    @overallLosses = 0
```

Comment:
If you never use a variable outside of a particular function, then you should make it local instead of an instance variable.